### PR TITLE
[REACT-5]: Implement menus and selects

### PR DESCRIPTION
**Pull request by LukeT**

### CODE REVIEW

File: packages/@streamjar/ui-react/src/lib/menu/menu.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	constructor(props: IMenuProps) {
+		super(props);
+
+		this.state = { hide: false };
+		this.menuRef = React.createRef();
+		this.close = this.close.bind(this);
+		this.renderMenu = this.renderMenu.bind(this);
+	}
+
+	public componentWillReceiveProps(prev: IMenuProps, next: IMenuProps): void {
```
Comment (EthanWaite) [inline packages/@streamjar/ui-react/src/lib/menu/menu.tsx]: this is deprecated

File: packages/@streamjar/ui-react/src/lib/select/select.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+		const classes = classnames({
+			'jar-select': true,
+			'jar-select-multiple': multiple,
+			'layout-row': true,
+		});
+
+		this.values = new Map();
+
+		const childs = React.Children.map(children, (child: any) => {
```
Comment (EthanWaite) [inline packages/@streamjar/ui-react/src/lib/select/select.tsx]: You caaaan cast this but it's a bit messy. My attempt's in the overlay code

File: packages/@streamjar/ui-react/src/lib/menu/menu.tsx [hunk 3]
```diff
@@ -39,12 +39,10 @@ export class Menu extends React.PureComponent<IMenuProps, IMenuState> {
 
 		this.state = { hide: false };
 		this.menuRef = React.createRef();
-		this.close = this.close.bind(this);
-		this.renderMenu = this.renderMenu.bind(this);
 	}
 
-	public componentWillReceiveProps(prev: IMenuProps, next: IMenuProps): void {
-		if (prev.anchor !== next.anchor) {
+	public componentWillUpdate(prev: IMenuProps): void {
```
Comment (EthanWaite) [inline packages/@streamjar/ui-react/src/lib/menu/menu.tsx]: this method is still deprecated. :P https://reactjs.org/docs/react-component.html#updating

Comment (LukeT) [reply packages/@streamjar/ui-react/src/lib/menu/menu.tsx]: fixed in a later PR 
image

Comment (EthanWaite) [reply packages/@streamjar/ui-react/src/lib/menu/menu.tsx]: ok

**Pull request merged**